src/app/widget-display/widget-display.component.ts

```diff
@@ -64,6 +75,7 @@ export class WidgetDisplayComponent implements OnInit, OnDestroy {
   closeWidget(index: number): void {
     this.widgets.splice(index, 1);
     this.updateButtonVisibility();
+    this.saveWidgets();
   }
 
   nextWidget(): void {
```

Comment:
nextWidget(): void {
  this.currentIndex = (this.currentIndex + 1) % this.widgets.length;
  this.updateButtonVisibility();
}

Reply:
prevWidget(): void {
  this.currentIndex = (this.currentIndex - 1 + this.widgets.length) % this.widgets.length;
  this.updateButtonVisibility();
}

Reply:
but it's not critical.